### PR TITLE
Add per-update decay_factor (forgetting) to bandit models

### PR DESCRIPTION
**Pull request by shaharbar1**
### Changes:

* Add decay_factor field to Model in pybandits/model/base.py — per-update forgetting factor in (0, 1]; a leaf-model attribute (Beta, BayesianNeuralNetwork), absent on quantitative wrappers
* Add _prior_pseudo_count ClassVar to BaseModelSO in pybandits/base_model.py — Beta(1, 1) anchor reused for n_successes/n_failures defaults and reset()
* Add decay logic to BaseBeta in pybandits/model/beta.py — decayed_n_successes/decayed_n_failures effective counts seeded by _init_decayed_counts validator; _update() discounts toward the prior (n <- 1 + decay_factor*(n-1) + new); sample_proba() draws from the effective counts when decay is enabled
  - register decayed counts in _transfer_learned_keys so they transfer alongside the raw counts
* Add decay_factor param to BaseBetaMO.cold_start() — forwarded to each per-objective Beta
* Add _inflate_prior_variance() to BaseBayesianNeuralNetwork in pybandits/model/bnn/network.py — scales weight/bias/embedding sigma by 1/decay_factor before each re-fit so fresh data dominates; leaves means (mu) and Student-t nu untouched
  - add decay_factor param to BaseBayesianNeuralNetwork.cold_start() and BaseBayesianNeuralNetworkMO.cold_start()
* Refactor BaseZooming in pybandits/quantitative_model/zooming.py — replace private _base_model/_init_base_model with a serialized base_model: Beta field built from base_model_cold_start_kwargs
* Update BaseQuantitativeBayesianNeuralNetwork.cold_start() in pybandits/quantitative_model/bnn.py — forward decay via base_model_cold_start_kwargs and default it to None
* Update QuantitativeModel docstring in pybandits/quantitative_model/base.py — document the shared base_model_cold_start_kwargs contract
* Update BaseMab.cold_start() docstring in pybandits/mab.py — note decay_factor is a forwardable action-model kwarg
* Add TestBetaDecayFactor to tests/test_model.py — seeding, raw counts stay pristine, decay toward prior, factor==1 tracks raw, sampling from effective counts, geometric decay on empty updates, reset, serialization round-trip, pre-decay state loads disabled, out-of-range rejection
* Add TestBnnDecayFactor to tests/test_model.py — weight/bias/embedding sigma scaled by 1/decay_factor with means unchanged
* Update tests/test_cmab.py — route decay_factor for BNN (top-level) and QuantitativeBNN (via base_model_cold_start_kwargs)
* Update tests/test_smab.py — route Zooming decay_factor through base_model_cold_start_kwargs; thread decay_factor through the create/cold-start helpers

### Changes:
* Feature or fix description
* Tests added or updated
* Documentation updated (if needed)

## Summary by CodeRabbit

* **New Features**
  * Added an optional `decay_factor` parameter to cold-start for supported models (including Beta-based, Bayesian neural network, and multi-objective/quantitative wrappers), enabling per-update forgetting that gradually shifts effective evidence back toward the prior.
* **Tests**
  * Expanded coverage for decay behavior, edge cases (e.g., `decay_factor=1`), reset/serialization compatibility, and invalid `decay_factor` validation.
* **Documentation / Chores**
  * Improved parameter documentation and bumped package version to 7.4.0.

### CODE REVIEW

File: pybandits/model/beta.py [hunk 1]
```diff
@@ -48,8 +51,33 @@ class BaseBeta(Model, ABC):
...
+        """Seed the effective decayed counts from the raw counts when decay is enabled."""
+        if self.decay_factor is not None:
+            if self.decayed_n_successes is None:
+                self.decayed_n_successes = float(self.n_successes)
+            if self.decayed_n_failures is None:
+                self.decayed_n_failures = float(self.n_failures)
+        return self
+
     @property
     def std(self) -> float:
```
Comment (AnastasiiaKabeshova) [inline pybandits/model/beta.py:82]: should std use effective counts when decay active?
decayed_n_failures and decayed_n_successes instead self.count

File: pybandits/model/bnn/network.py [hunk 2]
```diff
@@ -1506,6 +1510,30 @@ def _update(self, context: np.ndarray, rewards: List[BinaryReward]):
             updated_layer_params_list  # update the model_params with the new found posteriors
         )
 
+    def _inflate_prior_variance(self):
```
Comment (AnastasiiaKabeshova) [inline pybandits/model/bnn/network.py:1516]: It fires before every SVI fit
but I think it should not before the very first one.

**Review by AnastasiiaKabeshova — APPROVED** [review]
